mailpoet/lib/Automation/Engine/Validation/WorkflowRules/AtLeastOneTriggerRule.php

```diff
@@ -25,6 +25,11 @@ public function visitNode(Workflow $workflow, WorkflowNode $node): void {
   }
 
   public function complete(Workflow $workflow): void {
+    // run full step validation only for active workflows
+    if ($workflow->getStatus() !== Workflow::STATUS_ACTIVE) {
```

Comment:
should we also validate `Workflow::STATUS_DEACTIVATING`. I think in the end no, because no new user can enter the workflow and the current runs are on a different version. But thinking out loud, maybe I miss something.

Reply:
That's a good point. This applies to many of the rules. I think I could add a method on the workflow itself, something like `$workflow->needsFullValidation()` or some better name, if anyone has a better idea. Then it would all be in one place.